src/components/Create/useCreateLink.tsx

```diff
@@ -224,7 +224,10 @@ export const useCreateLink = () => {
             let transactionCostWei = feeOptions.gasLimit.mul(feeOptions.maxFeePerGas || feeOptions.gasPrice)
             let transactionCostNative = ethers.utils.formatEther(transactionCostWei)
             const nativeTokenPrice = await utils.fetchTokenPrice('0x0000000000000000000000000000000000000000', chainId)
-            const transactionCostUSD = Number(transactionCostNative) * nativeTokenPrice?.price
+            if (!nativeTokenPrice) {
+                throw new Error('Failed to fetch token price')
+            }
+            const transactionCostUSD = Number(transactionCostNative) * nativeTokenPrice.price
```

Comment:
suggestion: Can we use bigint here? I'm terrified of JS number precision :melting_face:

Reply:
could be done, but we must have into consideration that the price is not an int, so scaling must be used